django/contrib/auth/views.py

```diff
@@ -309,7 +309,22 @@ def get_user(self, uidb64):
             UserModel.DoesNotExist,
             ValidationError,
         ):
-            user = None
+            pass
+
+        # Temporarily support the old format of uid base64-encoded as plain text
```

Comment:
This is good, supporting old links. But we can't have this code forever in Django, we need a clear and documented deprecation path. My advice is to follow the [deprecating a feature docs](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#deprecating-a-feature) and raise a deprecation warning for each successful hit for old-style links.

I do wonder if we need a more clever approach here to handle the deprecation... There is an [ongoing conversation](https://forum.djangoproject.com/t/signing-the-csrf-cookie/35156/5) about signing CSRF cookies which has some common points with this. I will reference this PR in the forum topic but at this point I think we need to involve the community to design a good plan for this.

Reply:
Also, whatever we end up doing for the deprecation/transition of link and token obfuscation, we need to add tests for it.